src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -79,6 +79,9 @@ public class ConsensusManager : IConsensusManager
         /// <inheritdoc />
         public IConsensusRuleEngine ConsensusRules { get; private set; }
 
+        /// <inheritdoc />
+        public BestChainIndexer BestChainIndexer { get; }
```

Comment:
why should it be on CM's interface and not just a standalone component accessible via DI?

Reply:
It may be considered an helper class of ConsensusManager.
If we leave it this way, than this mean that my comment above about leaving blocksByHeight readonly can be true, instead of calling explicitly `this.BestChainIndexer.Initialize(pendingTip);` we can pass the pendingtip to BestChainIndexer constructor and so remove BestChainIndexer.Initialize method
oh, so actually we instantiate it

Reply:
I agree this was intended to be a helper class, sort of an extention to CM, however I realize sever methods can be made as external methods to CM, this may reduce the size of this class maybe we can then just put the code in CM itself.

Reply:
The dictionary `blocksByHeight` is only used by to methods, I hpw ever agree that BestChainIndexer could become a sort of helper class to CM even an extension to its functionality in which case making it a property of CM is not a bad idea.  to put in DI creates complications.